parser.py

```diff
@@ -0,0 +1,34 @@
+from calculator import Calculator
+from data_structures import WeatherData
+import utils
+from datetime import datetime
+
+
+class Parser:
+    
+    @classmethod
+    def parse_file_data(cls, file_data, file_name,weather_data):
+        for line in file_data[2:]:
+            values = line.replace('\n', '').split(',')[1:]
+            day_data = [datetime.strptime(line.split(',')[0], '%Y-%m-%d')]
+            for value in values:
+                cls.parse_value(value, day_data)
+
+            weather_data.append(WeatherData(file_name, day_data))
```

Comment:
return the weather_data here and add or append that in the function it is being called from